pkg/applicationserver/applicationserver.go

```diff
@@ -105,7 +105,27 @@ func (as *ApplicationServer) Context() context.Context {
 	return as.ctx
 }
 
-var errListenFrontend = errors.DefineFailedPrecondition("listen_frontend", "failed to start frontend listener `{protocol}` on address `{address}`")
+var (
+	errListenFrontend = errors.DefineFailedPrecondition(
+		"listen_frontend", "failed to start frontend listener `{protocol}` on address `{address}`",
+	)
+	errMaxRetriesReached = errors.DefineAborted(
+		"max_retries_reached", "maximum number of retries ({max_retries}) reached",
+	)
+)
+
+func validateDownlinkConfirmationConfig(c ConfirmationConfig) error {
+	if c.DefaultRetryAttempts == 0 && c.MaxRetryAttempts == 0 {
```

Comment:
We should add default configuration for these settings. I would say that by default 8 retries is more than sufficient to avoid people complaining about backwards incompatibility, and 32 as a max should be fine.

Also I would say that both values are required to be non zero.

Reply:
This is not solved, but I can see the confusion. Neither of the values should be zero (`||` not `&&`).